sdk/cosmosdb/cosmos/src/documents/PartitionKey.ts

```diff
@@ -1,5 +1,50 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
-import { PartitionKeyDefinition } from "./PartitionKeyDefinition";
 
-export type PartitionKey = PartitionKeyDefinition | string | number | unknown;
+import { NonePartitionKeyLiteral, NullPartitionKeyLiteral } from "./PartitionKeyInternal";
+
+export type PartitionKey = PrimitivePartitionKeyValue | PrimitivePartitionKeyValue[];
+
+/**
+ * A primitive Partition Key value.
+ */
+export type PrimitivePartitionKeyValue =
+  | string
+  | number
+  | boolean
+  | NullPartitionType
+  | NonePartitionKey;
+
+/**
+ * The returned object represents a partition key value that allows creating and accessing items
+ * with a null value for the partition key.
+ */
+export type NullPartitionType = null;
+
+/**
+ * The returned object represents a partition key value that allows creating and accessing items
+ * without a value for partition key
+ */
+export type NonePartitionKey = {
```

Comment:
I'm not sure I follow this type. Would you mind sharing a usage sample where this type would be involved?

Reply:
None partition key is `json` type `{}`. But since in javascript `{}` is also syntax for creating a new object. ie. `new Object()`. Using `{}` as a literal value is error prone. For example, here equality operator will fail,
`const partitionKey = {};
if(partitionKey === {}) // will evaluate to false. since {} is creating a new object.

Now, since using {} as both type and literal is problematic, to provide typesafety following type checks for the shape of object.
`export type NonePartitionKey = {
  [K in any]: never;
};`